core/org.eclipse.cdt.core/src/org/eclipse/cdt/core/build/ICBuildConfiguration.java

```diff
@@ -79,9 +80,23 @@ default String getLaunchMode() {
 	 *
 	 * @return binary parser ids
 	 * @throws CoreException
+	 * @deprecated As of 10.??? replaced by {@link ICBuildConfiguration#getBinaryParserIds}
 	 */
+	@Deprecated(since = "7.5")
 	String getBinaryParserId() throws CoreException;
 
+	/**
+	 * Ids for the Binary Parsers to use when checking whether a file is a
+	 * binary that can be launched.
+	 *
+	 * @return binary parser ids
+	 * @throws CoreException
+	 * @since 7.5
+	 */
+	default List<String> getBinaryParserIds() throws CoreException {
```

Comment:
With a new major revision (CDT 11) this method shouldn't have a default anymore.